src/uwtools/scheduler.py

```diff
@@ -269,33 +272,28 @@ class LSF(JobScheduler):
         RequiredAttribs.QUEUE: "-q",
         RequiredAttribs.ACCOUNT: "-P",
         RequiredAttribs.WALLTIME: "-W",
-        RequiredAttribs.NODES: "-n",
+        RequiredAttribs.NODES: lambda x: f"-n {x}",
         RequiredAttribs.TASKS_PER_NODE: lambda x: f"-R span[ptile={x}]",
         OptionalAttribs.SHELL: "-L",
         OptionalAttribs.JOB_NAME: "-J",
         OptionalAttribs.STDOUT: "-o",
         OptionalAttribs.THREADS: lambda x: f"-R affinity[core({x})]",
+        OptionalAttribs.MEMORY: lambda x: f"-R rusage[mem={x}]",
     }
 
-    def pre_process(self):
+    def pre_process(self) -> Dict[str, Any]:
         items = self._data
-        items[self._map[OptionalAttribs.THREADS](items[OptionalAttribs.THREADS])] = ""
+        # LSF requires threads to be set (if None is provided, default to 1)
+        items[OptionalAttribs.THREADS] = items.get(OptionalAttribs.THREADS, 1)
+        memory = items.get(OptionalAttribs.MEMORY, "")
+        if memory not in NONEISH:
+            items[self._map[OptionalAttribs.MEMORY](items[OptionalAttribs.MEMORY])] = ""
         items[
             self._map[RequiredAttribs.TASKS_PER_NODE](
                 items[RequiredAttribs.TASKS_PER_NODE]
             )
         ] = ""
-        items[
-            f"-n {int(items[RequiredAttribs.TASKS_PER_NODE] * int(items[RequiredAttribs.NODES]))}"
-        ] = ""
-        return items
-
-    def select(self, items: Dict[str, Any]):
-        """select logic"""
-
-        nodes = items.get(RequiredAttribs.NODES, "")
-        tasks_per_node = items.get(RequiredAttribs.TASKS_PER_NODE)
-        items[RequiredAttribs.NODES] = int(nodes) * int(tasks_per_node)
+        items[RequiredAttribs.NODES] = int(items[RequiredAttribs.TASKS_PER_NODE] * int(items[RequiredAttribs.NODES]))
```

Comment:
Should `items[RequiredAttribs.TASKS_PER_NODE]` be a guaranteed `int` here?

Reply:
Does it matter? doesn't `int( )` take care of that?

Reply:
If it's a string, what happens in the multiplication step?

Reply:
good eye.  The `int( )` was misplaced.